manifests/reverse_proxy.pp

```diff
@@ -26,43 +26,57 @@
 
   Class['certs', 'certs::ca', 'certs::apache', 'certs::foreman_proxy'] ~> Class['apache::service']
 
-  apache::vhost { 'katello-reverse-proxy':
-    servername             => $certs::apache::hostname,
-    aliases                => $certs::apache::cname,
-    port                   => $port,
-    docroot                => '/var/www/',
-    priority               => '28',
-    ssl_options            => ['+StdEnvVars', '+ExportCertData', '+FakeBasicAuth'],
-    ssl                    => true,
-    ssl_proxyengine        => true,
-    ssl_proxy_ca_cert      => $certs::ca_cert,
-    ssl_proxy_machine_cert => $certs::foreman_proxy::foreman_proxy_ssl_client_bundle,
-    ssl_cert               => $certs::apache::apache_cert,
-    ssl_key                => $certs::apache::apache_key,
-    ssl_chain              => $certs::katello_server_ca_cert,
-    ssl_ca                 => $certs::ca_cert,
-    ssl_verify_client      => 'optional',
-    ssl_verify_depth       => 10,
-    ssl_protocol           => $ssl_protocol,
-    request_headers        => ['set X_RHSM_SSL_CLIENT_CERT "%{SSL_CLIENT_CERT}s"'],
-    proxy_pass             => [
-      {
-        'path'         => $path,
-        'url'          => $url,
-        'reverse_urls' => [$url],
-        'params'       => $proxy_pass_params,
-      }
-    ],
-    error_documents        => [
-      {
-        'error_code' => '500',
-        'document'   => '\'{"displayMessage": "Internal error, contact administrator", "errors": ["Internal error, contact administrator"], "status": "500" }\''
-      },
-      {
-        'error_code' => '503',
-        'document'   => '\'{"displayMessage": "Service unavailable or restarting, try later", "errors": ["Service unavailable or restarting, try later"], "status": "503" }\''
-      },
-    ],
-    *                      => $vhost_params,
+  $vhost_name = "katello-proxy-content"
+  $ports = $port ? {
+    Array        => $port,
+    Stdlib::Port => [$port],
+    default      => ['8443'],
+  }
+
+  $ports.each |Stdlib::Port $current_port| {
+    apache::vhost { "${vhost_name}:${current_port}":
+      servername             => $certs::apache::hostname,
+      aliases                => $certs::apache::cname,
+      port                   => $current_port,
```

Comment:
I'm planning to attach the pulpcore fragment to this, and want to avoid having pulpcore listen on 8443

with that said, I've been trying a different approach